gusto/timeloop.py

```diff
@@ -110,12 +111,13 @@ def run(self, t, tmax, pickup=False):
             if state.output.Verbose:
                 print("STEP", t, dt)
 
+            state.parameter_update()
             t += dt
             state.t.assign(t)
 
             with timed_stage("Apply forcing terms"):
                 self.forcing.apply((1-alpha)*dt, state.xn, state.xn,
-                                   state.xstar, mu_alpha=mu_alpha[0])
+                                   state.xstar, stage=stage[0])
```

Comment:
So this flag is actually selecting whether you are doing an implicit or explicit step of forcing?  Why not do that:

```
self.forcing.apply(...., explicit=True)
```

And then below

`self.forcing.apply(..., explicit=False)`

Then you don't need any of this action at a distance with magic "stage" variables saying how terms are modified.